sklearn/utils/estimator_checks.py

```diff
@@ -72,6 +72,36 @@ def _safe_tags(estimator, key=None):
     return _DEFAULT_TAGS
 
 
+def _raise_xfail(reason, request=None):
+    """Mark a check as a known failure
+
+    This function is mostly useful to be able to skip the test
+    if pytest is not installed. Outside of a scikit-learn pytest
+    session the SkipTest exception will allways be raised.
+
+    Parameters
+    ----------
+    reason : str
+        reason for the known failure
+    request: default=None
+        result of the pytest request fixture.
+    """
+    if not getattr(sys, "_is_pytest_session", False):
+        # outside of a scikit-learn pytest session always raise SkipTest
+        raise SkipTest('XFAIL ' + str(reason))
+    try:
+        import pytest
+        if request is None:
```

Comment:
can request be None ?

Reply:
Not in the way it's used now. You are right -- simplified this function.